app/routes.py

```diff
@@ -1,2 +1,280 @@
+from sqlalchemy import asc, desc
+import datetime
+from datetime import datetime, date, time, timezone
+from dotenv import load_dotenv
+import os
+import requests
+import json
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def is_complete_function(completed_at):
+        if completed_at is not None:
+            return True
+        else:
+            return False
+
+def post_message_to_slack(text, blocks=None):
+    requests.post('https://slack.com/api/chat.postMessage', 
+    headers={'Authorization': f"Bearer {os.environ.get('SLACK_BOT_TOKEN')}"},
+    data={'channel': f"{os.environ.get('SLACK_CHANNEL')}", 'text': text})
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def handle_tasks():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(title= request_body["title"],
+    description= request_body["description"],
+    completed_at= request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": {
+        "id": new_task.task_id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": is_complete_function(new_task.completed_at)}}), 201
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def tasks_index():
+    
+    sort_query = request.args.get("sort")
+    tasks_response = []
+    if sort_query == "asc":
+        tasks = Task.query.order_by(asc(Task.title))
+
+        for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+        return jsonify(tasks_response), 200
+
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(desc(Task.title))
+
+        for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+        return jsonify(tasks_response), 200
+
+    else:
+        tasks = Task.query.all()
+    
+        if tasks is None:
+            return jsonify(tasks_response), 200
+    
+        else:
+            for task in tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete_function(task.completed_at)
+                })
+            return jsonify(tasks_response), 200
```

Comment:
👍  this works great! But we are running the same for loop three times, so let's try this:

```suggestion
    tasks_response = []
    if sort_query == "asc":
        tasks = Task.query.order_by(asc(Task.title))
    eliif sort_query == "desc":
        tasks = Task.query.order_by(desc(Task.title))
    else:
        tasks = Task.query.all()

    for task in tasks:
         tasks_response.append({
            "id": task.task_id,
            "title": task.title,
            "description": task.description,
            "is_complete": is_complete_function(task.completed_at)
         })
    return jsonify(tasks_response), 200
```